libs/keystore/fs_keystore.go

```diff
@@ -0,0 +1,126 @@
+package keystore
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+// fsKeystore implements persistent Keystore over OS filesystem.
+type fsKeystore struct {
+	path string
+}
+
+// NewFSKeystore creates a new Keystore over OS filesystem.
+// The path must point to a directory. It is created automatically if necessary.
+func NewFSKeystore(path string) (Keystore, error) {
+	err := os.Mkdir(path, 0755)
+	if err != nil && !os.IsExist(err) {
+		return nil, fmt.Errorf("keystore: failed to make a dir: %w", err)
+	}
+	return &fsKeystore{path: path}, nil
+}
+
+func (f *fsKeystore) Put(n KeyName, pk PrivKey) error {
+	path := f.pathTo(n.Base32())
+
+	_, err := os.Stat(path)
+	if err == nil {
+		return fmt.Errorf("keystore: key '%s' already exists", n)
+	} else if !os.IsNotExist(err) {
+		return fmt.Errorf("keystore: check before writing key '%s' failed: %w", n, err)
+	}
+
+	data, err := json.Marshal(pk)
```

Comment:
Just noting: From a security perspective, we should make sure that the time to encode a private key does not depend on the particular private key content. Otherwise the timing might leak contents of the key. As this is for p2p keys only and storing the keys in plain text is probably a greater risk than timing attacks here.

Reply:
https://github.com/ethereum/go-ethereum/blob/master/accounts/keystore/plain.go#L48